DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/README.md

```diff
@@ -0,0 +1,368 @@
+    <port name="mem1_r" direction="r"/>
+  </interface>
+</global_mem>
+<global_mem name="2" max_bandwidth="0" config_addr="0x0" type="device private" allocation_type="host,shared">
+  <interface name="2" port="mem2_rw" type="agent" width="64" address="0x3000000000000" latency="16" latency_type="fixed" waitrequest="1" size="0x10000000"/>
+</global_mem>
+<global_mem name="3" max_bandwidth="0" config_addr="0x0" type="device private" allocation_type="host,shared">
+    <interface name="3" port="mem3_rw" type="agent" width="64" address="0x4000000000000" latency="16" latency_type="fixed" waitrequest="1" size="0x10000000"/>
+</global_mem>
+```
+
+### Avalon MMHost Bus Signals
+A separate Avalon signal bus is generated at the top-level device image for each kernel argument that has a unique buffer location specified for it. The signal bus has customized address bus size, data bus size etc. based on the properties specified in the C++ source file.
+
+A simplified description of the signals that constitute the Avalon signal bus is shown below. A detailed description can be found in the Avalon Interface Specifications Document (ID: 683091, release 22.3) in section 3.2 titled: Avalon Memory Mapped Interface Signal Roles: 
+
+| Signal | Description | Width
+| --- | --- | ---
+| `address` | Represents the byte address | 1 - 64 bits
+| `writedata` | Data for write transfers | 8, 16, 32, 64, 128, 256, 512, or 1024 bits
+| `readdata` | The readdata driven from the agent to the host in response to a read transfer | 8, 16, 32, 64, 128, 256, 512, or 1024 bits
+| `byteenable` | Enables one or more specific byte lanes during transfers on interfaces of width greater than 8 bits. Each bit in byteenable corresponds to a byte in writedata and readadata | 2, 4, 8, 16, 32, 64, or 128 bits
+| `waitrequest` | An agent asserts this waitrequest signal when unable to respond to read/write requests | 1 bit 
+| `readdatavalid` | Used for variable-latency pipelined read transfers, asserts that the readdata signal contains valid data | 1 bit
+| `burstcount` | Used to indicate the number of transfers in each burst | 1 - 11 bits
+
+### Buffer Location in Pointer Address Bits
+The compiler encodes certain information regarding the virtual address space in top bits of the 64 bits wide pointer address. Below is a table showing which bits correspond to what function
+
+| Bits | Function
+| --- | ---
+| 0 - 40 | Addressing within the memory system
+| 41 - 46 | Buffer Location information
+| 47 | Not Used
+| 48 - 61 | Used by runtime layer
+| 62 - 63 | Indicates whether a pointer is a global or local memory pointer
+
+Users do not need to encode this information themselves in most cases. The compiler automatically generates logic to embed this information from the buffer location specified on the pointer kernel argument in the source file.
+
+Why is this embedding needed?
```

Comment:
From the FD, add the section about when the user needs to embed this information themselves.